app/processing/siti-agri/handle-known-defects/cap-bps-penalties/recalculate-bps-penalties.js

```diff
@@ -0,0 +1,34 @@
+const recalculateBPSPenalties = (paymentRequest) => {
+  const schemeCodes = [...new Set(paymentRequest.invoiceLines.map(invoiceLine => invoiceLine.schemeCode))]
+
+  schemeCodes.forEach((schemeCode) => {
+    const invoiceLinesByScheme = paymentRequest.invoiceLines.filter(invoiceLine => invoiceLine.schemeCode === schemeCode)
+    let grossAfterPenalties = calculateGrossAfterPenalties(invoiceLinesByScheme)
+
+    if (grossAfterPenalties < 0) {
+      const p04Penalty = invoiceLinesByScheme.filter(invoiceLine => invoiceLine.description.match(/^P04/gm))[0]
```

Comment:
You could use `find` rather than `filter` then you wouldn't need the `[0]` on the end.